### ISSUE TITLE
Unable to run mos on Macos Sonoma 14.1.2

### ISSUE THREAD

**Issue by nkroker** [Caldis/Mos#628]
**Describe the bug | 问题描述**
#### I just installed the mos on my macOS Sonoma and while launching it gets into this issue where macOs won't let it run due to security

![Screenshot 2023-12-03 at 10 38 03 AM](https://github.com/Caldis/Mos/assets/26537166/ced55033-6e73-4872-a319-81c614a968ab)

**To Reproduce | 如何重现**
#### System requirements: 
OS: **MacOs Sonoma 14.1.2**

#### Steps to reproduce
- Download and install
    - Install using brew `brew install mos`
    - Install via download https://mos.caldis.me/
    - Launch after installation
    - it will give this error
    **“Mos” can’t be opened because Apple cannot check it for malicious software.**

**Expected behavior | 期望结果**
> **It should run as usual**  
  
**Screenshots | 相关截图**

![Screenshot 2023-12-03 at 10 29 42 AM](https://github.com/Caldis/Mos/assets/26537166/7ec9289b-230e-4ca1-8297-83439773c32d)

[![Video of the bug](https://github.com/Caldis/Mos/assets/26537166/3d05df6a-dfa5-4e5a-900a-248630290dfd)](https://github.com/Caldis/Mos/assets/26537166/f800a9d2-0b30-47af-b8a9-f23af88b6c73)

**System Info | 系统信息:**
 - Mouse: **Portronics Toad One**
 - OS: **MacOS Sonoma 14.1.2**
 - Browser: **Chrome**
 - Version: **Mos.Versions.3.4.1**
 -  Application: **Whole System**

**Additional context | 额外说明**
I also have **[xGestures](https://briankendall.net/xGestures/)** installed on my system to mimic the trackpad and magic mouse gestures but I know **mos** is not even starting that's the problem.

**Comment by VladimirCo** [comment]
> I think this is the same issue can we link both of them #606

**Why does this message appear?**

In macOS, there is a security called 'Notarization'. When an app is notarized, that means Apple has checked it and determined it to be safe.
Unfortunately, it costs $99 to notarize an app, making it a suboptimal option for small open-source projects like Mos.
So Mos is not notarized, that's why this message appears.

However, since the Mos source code is freely available and thousands of people are using it and have the ability to write about their experiences on this GitHub page, you can still be sure that Mos is safe.

**The solution**

To open Mos anyways:

 1. Right-click Mos in the Finder, then click 'Open'
 2. Click 'Open' again in the window that appears
 
Mos should then open normally and the 'cannot check for malicious software' message won't appear anymore.

**Comment by DavidPal02** [comment]
I tried to open with Finder but still appear the windows to send the app to trash

**Comment by seclerp** [comment]
The workaround doesn't work on MacOS 15

<img width="372" alt="image" src="https://github.com/user-attachments/assets/df099111-8954-4c41-be82-1484ab2b9ca0">

**Comment by seclerp** [comment]
According to [the guide](https://www.youtube.com/watch?v=biIvAM94b98), a new workaround that works on MacOS Sequoia (15) is the following:

1. Try to open Mos, then click Done
2. Open System Settings | Privacy & Security
3. Scroll down to the Security section
4. Click Open anyway near the text about Mos could not be verified and blah-blah

After that, I was finally able to open Mos without any issues.

**Comment by Murilo-Perrone** [comment]
@seclerp yes, but this workaround is not working here for Somona 14.3. The "Open anyway" button won't work:

<img width="488" alt="image" src="https://github.com/user-attachments/assets/211e0386-63e5-4dd0-88d3-fe3232fd459b">

PS: Im using a corporately managed machine.

**Comment by nkroker** [comment]
> @seclerp yes, but this workaround is not working here for Somona 14.3. The "Open anyway" button won't work:
> 
> <img alt="image" width="488" src="https://private-user-images.githubusercontent.com/26028390/375033301-211e0386-63e5-4dd0-88d3-fe3232fd459b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyMzc0NTMsIm5iZiI6MTcyOTIzNzE1MywicGF0aCI6Ii8yNjAyODM5MC8zNzUwMzMzMDEtMjExZTAzODYtNjNlNS00ZGQwLTg4ZDMtZmUzMjMyZmQ0NTliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDA3MzkxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1ODFjOTNhMmYzYTRjODYwMjg3MTk5MDdmOTU0NTk2ODk0OWYyZTRjMGU4ZjJlZWUwZTRhYjIyNWFjZDBiN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DViECeyntBW12VKcD7Oej_eVvEE3W6dqZfD4T81IBbw">
> PS: Im using a corporately managed machine.

There are two ways to do this, 
- One way to change the MacOS setting from recovery to allow running applications from unidentified developers by changing the security levels.

- Self sign the app file iself by using this command
`sudo codesign -f -s - --deep "FILEPATH"`
replace the FILEPATH with the mos installation file

**Comment by gochev** [comment]
sudo codesign -f -s - --deep "FILEPATH"

doesnt work still the same ;)

**Comment by vishnusreddy** [comment]
> According to [the guide](https://www.youtube.com/watch?v=biIvAM94b98), a new workaround that works on MacOS Sequoia (15) is the following:
> 
> 1. Try to open Mos, then click Done
> 2. Open System Settings | Privacy & Security
> 3. Scroll down to the Security section
> 4. Click Open anyway near the text about Mos could not be verified and blah-blah
> 
> After that, I was finally able to open Mos without any issues.

This works, and I guess we should add this in the docs/troubleshooting steps cos it impacts every fresh installation.

**Comment by gochev** [comment]
@vishnusreddy  most macs use company policy so this doesnt work you cannot change stuff arround.


P.S. a workaround is to simply download the source of MOS and build it on your own then it works :>
